activerecord/lib/active_record/associations/through_association.rb

```diff
@@ -17,7 +17,7 @@ def target_scope
             scope.merge!(
               reflection.klass.all.
                 except(:select, :create_with, :includes, :preload, :joins, :eager_load)
-            )
+            ) if reflection.table_name != scope.table.name
```

Comment:
Avoiding inverse conditionals in favor of the more explicit syntax would be preferable.